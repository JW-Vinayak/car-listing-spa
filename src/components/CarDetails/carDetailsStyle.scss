section.car-details {
  height: $fixed-height;
  .image-container {
    @include off-white-border-bottom;
    text-align: center;
  }

  section {
    margin-top: $margin-24;
    display: flex;
    flex-wrap: wrap;
    justify-content: space-evenly;
    .car-info {
      width: 50%;
      min-width: 350px;
      margin-bottom: $margin-24;

      .car-name {
        font-size: 2em;
        margin-bottom: $margin-24;
      }

      .car-info {
        font-size: 1.2em;
        margin-bottom: $margin-24;
      }
    }

    .favorite-car {
      width: 30%;
      min-width: 300px;
      min-height: 130px;
      padding: $padding-24;
      position: relative;
      @include off-white-border;
    }
  }
}
