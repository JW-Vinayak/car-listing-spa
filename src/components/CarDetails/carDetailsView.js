import React from "react";
import { getCarDetails } from "../../utils/api";
import SetFavoriteCarView from "./FavoriteCar/favoriteCarView";
import PropTypes from "prop-types";

class CarDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      carDetails: null,
      error: false
    };
  }

  componentDidMount() {
    getCarDetails(this.props.match.params.stockNumber)
      .then(details => {
        this.setState({
          carDetails: details
        });
      })
      .catch(e => {
        this.setState({
          error: true
        });
      });
  }

  render() {
    if (this.state.error) {
      return (
        <div className="error error-box">
          An error occurred while fetching car details, please try again after
          some time.
        </div>
      );
    }
    if (!this.state.carDetails) {
      return <div className="error-box">Data is Loading...</div>;
    }
    let car = this.state.carDetails;
    return (
      <section className="car-details">
        <div className="image-container">
          <img
            alt={"car, " + car.modelName}
            className="car-image-big"
            src={car.pictureUrl}
          />
        </div>
        <section>
          <div className="car-info">
            <h1 className="car-name">{car.modelName}</h1>
            <div className="car-info">
              <span>Stock # {car.stockNumber} - </span>
              <span>
                {car.mileage.number} {car.mileage.unit.toUpperCase()} -{" "}
              </span>
              <span>{car.fuelType} - </span>
              <span>{car.color}</span>
            </div>
            <p>
              This car is currently available and can be delivered as soon as
              tomorrow morning.Pleaese be aware that delivery times shown in
              this page are not definitive and may change due to bad weather
              conditions.
            </p>
          </div>
          <div className="favorite-car">
            <SetFavoriteCarView {...car} />
          </div>
        </section>
      </section>
    );
  }
}

CarDetails.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.object.isRequired
  })
};

export default CarDetails;
