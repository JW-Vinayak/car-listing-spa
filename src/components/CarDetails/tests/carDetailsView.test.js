import React from "react";
import CarDetails from "../carDetailsView";
import renderer from "react-test-renderer";

let match = {
  params: {
    stockNumber: 1234
  }
};

describe("Car details component", () => {
  it("shallow renders without crashing", () => {
    const tree = renderer.create(<CarDetails match={match} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
