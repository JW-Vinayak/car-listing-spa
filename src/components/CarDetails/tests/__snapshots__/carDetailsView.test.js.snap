// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Car details component shallow renders without crashing 1`] = `
<div
  className="error-box"
>
  Data is Loading...
</div>
`;
